File: src/net/eduard/api/server/currency/list/CurrencyEduCash.java
package net.eduard.api.server.currency.list;

import net.eduard.api.lib.game.FakePlayer;
import net.eduard.api.server.Systems;
import net.eduard.api.server.currency.SimpleCurrencyHandler;

public class CurrencyEduCash extends SimpleCurrencyHandler {

    public CurrencyEduCash(){
        setName("EduCash");
        setDisplayName("Sistema de Cash");

    }

    @Override
    public double get(FakePlayer player) {
        return Systems.getCashSystem().getCash(player);
    }


    @Override
    public boolean remove(FakePlayer player, double amount) {

        Systems.getCashSystem().removeCash(player, amount);

        return true;
    }

    @Override
    public boolean check(FakePlayer player, double amount) {

        return Systems.getCashSystem().hasCash(player, amount);

    }

    @Override
    public boolean add(FakePlayer player, double amount) {

        Systems.getCashSystem().addCash(player, amount);
        return true;

    }
}

File: src/main/java/net/eduard/api/supports/CurrencyEduCash.kt
package net.eduard.api.supports

import net.eduard.api.lib.storage.annotations.StorageAttributes
import net.eduard.api.server.currency.SimpleCurrencySystem
import net.eduard.api.lib.modules.FakePlayer
import net.eduard.api.server.Systems

@StorageAttributes(indentificate = true)
class CurrencyEduCash : SimpleCurrencySystem() {
    override operator fun get(player: FakePlayer): Double {
        return Systems.cashSystem!!.getCash(player)
    }

    override fun contains(player: FakePlayer, amount: Double): Boolean {
        return Systems.cashSystem!!.hasCash(player, amount)
    }

    override fun remove(player: FakePlayer, amount: Double): Boolean {
        Systems.cashSystem!!.removeCash(player, amount)
        return true
    }


    override fun add(player: FakePlayer, amount: Double): Boolean {
        Systems.cashSystem!!.addCash(player, amount)
        return true
    }

    init {
        name = "EduCash"
        displayName = "Sistema de Cash"
    }
}